test/files/rules/quotemark-allow-escape.test.ts

```diff
@@ -0,0 +1,2 @@
+var single = "'single'";
+    var doublee = '"doublee"';
```

Comment:
I'd prefer if these test cases were just added to `quotemark.test.ts`. That way, we can verify that running the rule _without_ the `"escape"` option catches these as failures. It also means that you'll have to modify some existing tests.